artifact-app/build.rs

```diff
@@ -53,8 +55,22 @@ fn check_deps() {
 
 fn build_mdbook() {
     println!("Building the book");
+
+    let mdbook_dir = if cfg!(windows) {
+
+        let regex = Regex::new("[a-zA-Z].+").unwrap();
+        let shortened = regex.find(BOOK.as_path().to_str().expect("Invalid book dir"))
+            .unwrap();
+        &BOOK.as_path()
+            .to_str()
+            .expect("Invalid book dir")[shortened.start() .. shortened.end()]
```

Comment:
I'm still unsure how this equates to what you want. Does this really work for all of these case?

```
C:/foo
C:/foo/bar
//?/C:foo/bar/baz
```

I think you are trying to remove the leading `//?/`, but wouldn't you want to slice from `book[shortened.end() .. ]`?

The fact that windows doesn't accept it's own absolute paths is very concerning, and suggests to me that `path_abs` (a crate I maintain in `ergo`) should handle this case. Maybe `PathInfo::to_non_absolute()` or some such method.

Reply:
See regex tests here for demonstration: https://regex101.com/r/pOevIF/1
The slice could be changed to `[shortened.start .. ]`, which is more concise and essentially the same since the end of the matched text will always be the end of the string, `.+` matching any number of any character - it's a matter of style so it's up to you.

Reply:
On the topic of the windows API - everything in the windows API is concerning. I think the issue here, however, is likely in std::process::Command. I need to put more time into investigating this, but if the issue is in the standard library here it makes more sense for that to be fixed rather than having to handle strange platform edge cases in your own library.

Reply:
somehow I missed the `.`. Makes sense now. Either is fine.